src/api/handlers/webhook.ts

```diff
@@ -227,27 +227,69 @@ export function getSlashCommands(guildID?: string) {
 /**
  * Edit an existing slash command. If this command did not exist, it will create it.
  */
-export function upsertSlashCommand(options: UpsertSlashCommandOptions) {
-  return RequestManager.post(
-    options.guildID
-      ? endpoints.COMMANDS_GUILD_ID(applicationID, options.id, options.guildID)
-      : endpoints.COMMANDS_ID(applicationID, options.id),
-    {
-      ...options,
-    },
+export function upsertSlashCommand(
+  commandID: string,
+  options: UpsertSlashCommandOptions,
+  guildID?: string,
+) {
+  // Use ... for content length due to unicode characters and js .length handling
+  if ([...options.name].length < 2 || [...options.name].length > 32) {
+    throw new Error(Errors.INVALID_SLASH_NAME);
+  }
+
+  if (
+    [...options.description].length < 1 || [...options.description].length > 100
+  ) {
+    throw new Error(Errors.INVALID_SLASH_DESCRIPTION);
+  }
+
+  const result = RequestManager.patch(
+    guildID
+      ? endpoints.COMMANDS_GUILD_ID(
+        applicationID,
+        guildID,
+        commandID,
+      )
+      : endpoints.COMMANDS_ID(applicationID, commandID),
+    options,
   );
+
+  return result;
 }
 
-/** Edit an existing slash command. */
-export function editSlashCommand(options: EditSlashCommandOptions) {
-  return RequestManager.patch(
-    options.guildID
-      ? endpoints.COMMANDS_GUILD_ID(applicationID, options.id, options.guildID)
-      : endpoints.COMMANDS_ID(applicationID, options.id),
-    {
-      ...options,
-    },
+// TODO: remove this function for v11
+/** 
+ * Edit an existing slash command. 
+ * @deprecated This function will be removed in v11. Use `upsertSlashCommand()` instead
```

Comment:
The endpoints still exists:
- https://discord.com/developers/docs/interactions/slash-commands#edit-global-application-command
- https://discord.com/developers/docs/interactions/slash-commands#edit-guild-application-command

Reply:
Yes these exist and are covered by `upsertSlashCommand` imo `upsertSlashCommand` has a more descriptive name of what it is doing so that one should be kept.
Also theoretically `editSlashCommand` could be removed already without being a real breaking change since nobody uses it.

Reply:
> Yes these exist and are covered by upsertSlashCommand imo upsertSlashCommand has a more descriptive name of what it is doing so that one should be kept.
The usage of editSlashCommand() is different from that of upsertSlashCommand(), what if the user does not want to edit the slash command if the command does not exist? `upsertSlashCommand()` will create the command regardless and is a hacky solution... we should support both of the functions.
> ...since nobody uses it.
How do you know that?

Reply:
Well `editSlashCommand` is broken if someone would use that he couldn't and therefor would either ask for help or just not use it.

Reply:
> Well `editSlashCommand` is broken if someone would use that he couldn't and therefor would either ask for help or just not use it.

Even if the handler has a bug does not mean that we should just remove it, does it?

Reply:
Well rn you cannot do anything with this function. Rn it only accepts the `commandID` and the `guildID` as option but nothing that could be used to edit the command.

Reply:
You can add the new properties to the `EditSlashCommandOptions` and not deprecate the handler as well as interface.

Reply:
I have done that. In this PR `editSlashCommand` does work. But it is completely unnecessary due to `upsertSlashCommand` therefore it is redundant and should be removed.

Reply:
As I said, they both have different usage.

Reply:
No, they both have the exact same code and do the same request.

Reply:
There is no `edit` according to the discord documentation. If the command exists it gets edited if not gets created. That's what `upsertSlashCommand` and `editSlashCommand` does. `upsertSlashCommand` has a better more descriptive name so this should be kept and `editSlashCommand` should be removed.

Reply:
Hmm, just took a look at the code, I seem to be mistaken, but I think editSlashCommand() is a better-suiting name than upsertSlashCommand(), therefore, you should deprecate the `upsertSlashCommand()~ instead of `editSlashCommand()`.

Reply:
> Because Slash Commands have unique names within a scope, we treat POST requests for new commands as `upserts`